R/readutils.R

```diff
@@ -227,72 +243,65 @@ readcmifiles <- function(files, excludelist=c(""), skip, verbose=FALSE,
   # when stride is > 1, we only read a subset of files
   nFilesToLoad <- as.integer(nFiles/stride)
   nCols <- length(tmpdata)
-  ## we generate the full size data.frame first
-  tmpdata[,] <- NA
-  ldata <- tmpdata
-  ldata[((nFilesToLoad-1)*fLength+1):(nFilesToLoad*fLength),] <- tmpdata
-  # create a progress bar
   
-  pb <- NULL
-  if(!verbose) {
-    pb <- txtProgressBar(min = 0, max = nFiles, style = 3)
-  }
-  for(i in c(1:nFiles)) {
-    # update progress bar
-    if(!verbose) {
-      setTxtProgressBar(pb, i)
-    }
-    if( !(files[i] %in% excludelist) && file.exists(files[i]) && (i-1) %% stride == 0) {
-      
-      if(verbose) {
-        message("Reading from file ", files[i], "\n")
-      }
-      ## read the data
-      tmpdata <- read.table(files[i], colClasses=colClasses, skip=skip)
-      if(reduce) {
-        tmpdata <- tmpdata[tmpdata[,obs.index] %in% obs,]
-      }
-      ## sanity checks
-      if(fLength < length(tmpdata$V1)) {
-        warning(paste("readcmifiles: file ", files[i], " does not have the same length as the other files. We will cut and hope...\n"))
-      }
-      else if(fLength > length(tmpdata$V1)) {
-        stop(paste("readcmifiles: file", files[i], "is too short. Aborting...\n"))
+  dat <- my_lapply(
```

Comment:
@urbach if you're curious, this pattern of iterating in parallel using `mclapply` (or in this case `pbmclapply`) over the files and binding the results together with `do.call(rbind,dat)` in line 281 below is about a factor of 100 faster than building the data frame ahead of the time and replacing it bit by bit with the read data

Reply:
Even single-threaded, it is many many times faster.